pkg/target/target_flags.go

```diff
@@ -69,19 +73,24 @@ func (tf *targetFlagsImpl) ShootName() string {
 	return tf.shootName
 }
 
+func (tf *targetFlagsImpl) ControlPlane() bool {
+	return *tf.controlPlane
+}
+
 func (tf *targetFlagsImpl) AddFlags(flags *pflag.FlagSet) {
 	flags.StringVar(&tf.gardenName, "garden", "", "target the given garden cluster")
 	flags.StringVar(&tf.projectName, "project", "", "target the given project")
 	flags.StringVar(&tf.seedName, "seed", "", "target the given seed cluster")
 	flags.StringVar(&tf.shootName, "shoot", "", "target the given shoot cluster")
+	flags.BoolVar(tf.controlPlane, "controlplane", *tf.controlPlane, "target control plane of shoot, use together with shoot argument")
 }
 
 func (tf *targetFlagsImpl) ToTarget() Target {
-	return NewTarget(tf.gardenName, tf.projectName, tf.seedName, tf.shootName)
+	return NewTarget(tf.gardenName, tf.projectName, tf.seedName, tf.shootName, *tf.controlPlane)
 }
 
 func (tf *targetFlagsImpl) isEmpty() bool {
-	return tf.gardenName == "" && tf.projectName == "" && tf.seedName == "" && tf.shootName == ""
+	return tf.gardenName == "" && tf.projectName == "" && tf.seedName == "" && tf.shootName == "" && !*tf.controlPlane
```

Comment:
We should consider it to be empty even if the controlPlane is true

Reply:
I was wondering the same

Reply:
.. same for https://github.com/gardener/gardenctl-v2/blob/3b506a41fc8e4459764f82026717e3f77ea98a8b/pkg/target/target.go#L183

Reply:
I changed this back and forth several times during development. There were some cases where the targeting did not work, e.g. `target --controlplane` without specifying other target params, however I was not able to reproduce it. Therefore I removed `controlplane` checks from the isEmpty functions. If we encounter issues we can re-add it.